lib/connection/connection.js

```diff
@@ -40,175 +36,242 @@ var debugFields = [
- * @param {boolean} [options.promoteBuffers=false] Promotes Binary BSON values to native Node Buffers.
+ * A class representing a single connection to a MongoDB server
+ *
  * @fires Connection#connect
  * @fires Connection#close
  * @fires Connection#error
  * @fires Connection#timeout
  * @fires Connection#parseError
- * @return {Connection} A cursor instance
+ * @fires Connection#message
  */
-var Connection = function(messageHandler, options) {
-  // Add event listener
-  EventEmitter.call(this);
-  // Set empty if no options passed
-  this.options = options || {};
-  // Identification information
-  this.id = _id++;
-  // Logger instance
-  this.logger = Logger('Connection', options);
-  // No bson parser passed in
-  if (!options.bson) throw new Error('must pass in valid bson parser');
-  // Get bson parser
-  this.bson = options.bson;
-  // Grouping tag used for debugging purposes
-  this.tag = options.tag;
-  // Message handler
-  this.messageHandler = messageHandler;
-
-  // Max BSON message size
-  this.maxBsonMessageSize = options.maxBsonMessageSize || 1024 * 1024 * 16 * 4;
-  // Debug information
-  if (this.logger.isDebug())
-    this.logger.debug(
-      f(
-        'creating connection %s with options [%s]',
-        this.id,
-        JSON.stringify(debugOptions(debugFields, options))
-      )
-    );
+class Connection extends EventEmitter {
+  /**
+   * Creates a new Connection instance
+   *
+   * @param {Socket} socket The socket this connection wraps
+   * @param {Object} [options] Optional settings
+   * @param {string} [options.host] The host the socket is connected to
+   * @param {number} [options.port] The port used for the socket connection
+   * @param {boolean} [options.keepAlive=true] TCP Connection keep alive enabled
+   * @param {number} [options.keepAliveInitialDelay=300000] Initial delay before TCP keep alive enabled
+   * @param {number} [options.connectionTimeout=30000] TCP Connection timeout setting
+   * @param {number} [options.socketTimeout=360000] TCP Socket timeout setting
+   * @param {boolean} [options.promoteLongs] Convert Long values from the db into Numbers if they fit into 53 bits
+   * @param {boolean} [options.promoteValues] Promotes BSON values to native types where possible, set to false to only receive wrapper types.
+   * @param {boolean} [options.promoteBuffers] Promotes Binary BSON values to native Node Buffers.
+   */
+  constructor(socket, options) {
+    super();
+
+    options = options || {};
+    if (!options.bson) {
+      throw new TypeError('must pass in valid bson parser');
+    }
+
+    this.id = _id++;
+    this.options = options;
+    this.logger = Logger('Connection', options);
+    this.bson = options.bson;
+    this.tag = options.tag;
+    this.maxBsonMessageSize = options.maxBsonMessageSize || 1024 * 1024 * 16 * 4;
+
+    this.port = options.port || 27017;
```

Comment:
makes me really want options management is core :(

Reply:
agreed. _soon_